File: 371/SumOfTwoIntegers.java
public class SumOfTwoIntegers {

    // public int getSum(int a, int b) {
    //     while (b != 0) {
    //         int carrier = (a & b) << 1;
    //         a ^= b;
    //         b = carrier;
    //     }
    //     return a;
    // }

    public int getSum(int a, int b) {
        return b == 0 ? a : getSum(a ^ b, (a & b) << 1);
    }

}

File: 371/sum-of-two-integers.py
class Solution:
    def getSum(self, a, b):
        """
        :type a: int
        :type b: int
        :rtype: int
        """
        if b == 0:
            return a
        else:
            return self.getSum(a ^ b, (a & b) << 1)

print(Solution().getSum(-1, 1))